pyproject.toml

```
@@ -31,6 +31,7 @@ version = "0.0.0"
 deltalake = ["deltalake"]
 sqlalchemy = ["sqlalchemy"]
 pyarrow = ["pyarrow"]
+pydantic-core = ["pydantic-core>=2"]
```

```suggestion
pydantic= ["pydantic>=2"]
```

The code probably runs with just core, but the user would want actual `pydantic`, right? It would not be helpful for them to `pip install dataframely[pydantic-core]` because they would still need `pydantic` to use the functionality.